### PR TITLE
Implement ext-ds generics

### PR DESCRIPTION
**Pull request by enumag**
Closes https://github.com/phpstan/phpstan/issues/2615

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
You need to reference the stub file in `config.neon`.

**Comment by ondrejmirtes** [comment]
Hi, do you plan to finish this? Thanks!

**Comment by enumag** [comment]
Of course, I'm just quite busy lately. :-(

File: stubs/ext-ds.php [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+	/**
+	 * @param iterable<TValue> $values
+	 * @return Deque<TValue>
+	 */
+	public function merge($values): Deque
+	{
+	}
+
+	/**
+	 * @param (callable(TValue): bool)|null $callback
```
Comment (dktapps) [inline stubs/ext-ds.php]: `@return` missing here

**Comment by ondrejmirtes** [comment]
Please open a new PR when you're ready to finish it :) Thanks!

**Comment by enumag** [comment]
I noticed in existing stubs that you sometimes use `@template` and sometimes `@template-covariant`. Can you explain the difference to me please?

**Pull request closed without merging**